src/odb/include/odb/db.h

```diff
@@ -4103,6 +4113,21 @@ class dbObstruction : public dbObject
   ///
   static void destroy(dbObstruction* obstruction);
 
+  ///
+  /// Returns true if this obstruction is virtual. Virtual obstructions
+  /// represent obstructions created by non-rectangular floorplans.
+  /// The general flow is the polygonal floorplan is subtracted
+  /// from its general bounding box and the shapes that are created
+  /// by that difference are then decomposed into rectangles which
+  /// create obstructions with the virtual annotation.
+  ///
+  bool isVirtual();
+
+  ///
+  /// Sets this obstruction as virtual.
+  ///
+  void setIsVirtual(bool is_virtual);
+
```

Comment:
this feels really hacky to me. If the issue is that the various tools dont support this, then we should create issues and assign them out to ensure the tools can handle this properly.

Reply:
I feel like this is a reasonably good solution that covers most tools requirements for awareness here both now and going forward.

Some tools will need to be more aware, but for most this seems sufficient. The only time virtual comes up as necessary is in the GUI and in the lef and def writers. Where we do take on some kludge.

But it feels like we more often deal with tools that will not care about than virtual annotation than those who do like LEF/DEF/GUI. All of which have rather minor additional if conditions.

Reply:
@gadfort I suspect most tools will wind up doing something similar internally as they naturally want a rectangle to work in.  The 'virtual' name isn't great though it is hard to come up with a nicer one.